### PR TITLE
Searchpage & redirection to bookpage

### PR DESCRIPTION
**Pull request by igovor01**
Worked on /search

Done: 

- fetch books and display them
- functional searchbar and results section
- dynamic routing to a unique book page

To be done: 
- pagination
- services/hooks, i'm a bit lost, i want to use useBooks in SearchDropdown as i did in SearchResults but i'm not sure exactly how

### CODE REVIEW

File: app/layout.tsx [hunk 1]
```diff
@@ -19,17 +19,10 @@ export default function RootLayout({
   children: React.ReactNode;
 }) {
   return (
-    <html lang="en">
-      <body className="flex flex-col h-screen justify-between">
+    <html lang="en" className="!scroll-smooth">
```
Comment (ilovric00) [inline app/layout.tsx:22]: hmm what is this? Disable smooth scroll?

File: app/search/page.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            {results && <SearchDropdown results={results} />}
+          </section>
+
+          <div className="mt-6">
+            <Image
+              height={450}
+              width={450}
+              alt="header picture"
+              src="/assets/pixeltrue-seo.svg"
+              className=""
```
Comment (ilovric00) [inline app/search/page.tsx:52]: Remove it

File: app/search/page.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  const [searchValue, setSearchValue] = useState("");
+  const [searchMode, setSearchMode] = useState<SearchMode>({
+    modeKeyword: "",
+    label: "Everything",
+  });
+  const [isSearchDropdownOpen, setSearchDropdownOpen] = useState(false)
+
+  const toggleResultSection = () => {
+    if (!isResultSectionVisible)
+      setResultSectionVisible(!isResultSectionVisible);
```
Comment (ilovric00) [inline app/search/page.tsx:23]: always use brackets `if {}`

File: components/ResultSection.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+}: ResultSectionProps) => {
+  const { books, isLoading, error } = useBooks(
+    searchMode,
+    searchValue,
+    MAX_RESULTS
+  );
+
+  return (
+    <section
+      id="resultSection"
```
Comment (ilovric00) [inline components/ResultSection.tsx:26]: Do you need this `id` selector? I can't find it to be used somewhere

File: components/ResultSectionCard.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            categories.map((category) => {
+              return (
+                <div
+                  key={category}
+                  className="inline-flex py-1.5 px-4 bg-orange-200 rounded-full text-center font-semibold text-xs mr-0.5 mt-1"
+                >
+                  {category}
+                </div>
+              );
+            })}
```
Comment (ilovric00) [inline components/ResultSectionCard.tsx:54]:
```
{categories?.map(category => (
  <div
    key={category}
    className="inline-flex py-1.5 px-4 bg-orange-200 rounded-full text-center font-semibold text-xs mr-0.5 mt-1"
  >
    {category}
  </div>
))}
```

File: components/SearchBar.tsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+          className="flex items-center justify-between text-base font-medium text-white py-4 px-6"
+        >
+          <span>{capitalizeFirstLetter(searchMode.label)}</span>
+          <FaChevronDown
+            className={
+              isOpen
+                ? "rotate-180 transition-all duration-500"
+                : "transition-all duration-500"
+            }
+          />
```
Comment (ilovric00) [inline components/SearchBar.tsx:125]:
```
<FaChevronDown
  className={
    `transition-all duration-500 ${isOpen && 'rotate-180'}`
  }
/>
```

File: components/SearchBar.tsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+        </div>
+
+        <ul
+          className={
+            "absolute top-16 left-0 w-full rounded-2xl bg-white text-left overflow-hidden " +
+            (isOpen
+              ? "max-h-72 transition-all duration-500"
+              : "max-h-0 transition-all duration-500")
```
Comment (ilovric00) [inline components/SearchBar.tsx:133]: same here

File: components/SearchDropdown.tsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+interface SearchDropdownProps {
+  results: BookInfo[];
+}
+
+export const SearchDropdown = ({ results }: SearchDropdownProps) => {
+  return (
+    <div className="w-full max-w-xl bg-white flex flex-col shadow-sm rounded-md mt-4 max-h-80 overflow-y-scroll absolute">
+      {results.map((result) => {
+        return <SearchDropdownCard key={result.id} {...result} />;
```
Comment (ilovric00) [inline components/SearchDropdown.tsx:13]: remove `return` word

**Review by ilovric00 — CHANGES_REQUESTED** [review]

File: app/layout.tsx (hunk 1 above)
Comment (igovor01) [reply app/layout.tsx:22]: It was... needed for something I tried to do but didn't end up implementing. I'll remove it.

File: hooks/useBooks.ts [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+            id: book?.id || "",
+            title: book?.volumeInfo?.title || "",
+            authors: book?.volumeInfo?.authors || ["Unknown"],
+            publishedDate: book?.volumeInfo?.publishedDate || "",
+            textSnippet: book?.searchInfo?.textSnippet || "",
+            thumbnail: imgSrc,
+            categories: book?.volumeInfo?.categories || ["Other"],
+            pageCount: book?.volumeInfo?.pageCount || null,
+          };
+        });
```
Comment (ilovric00) [inline hooks/useBooks.ts:66]: this would be good to refactor a bit

**Review by ilovric00 — APPROVED** [review]

**Pull request merged**